unfolding/1D/Bayes/runCorrectionChain1DBayes_SpectrumTaskSimplePoor_8TeV.cpp

```diff
@@ -0,0 +1,446 @@
+    eff->SetDirectory(nullptr);
+    eff->Divide(trgspec.get(), mbref.get(), 1., 1., "b");
+    return eff;
+}
+
+TH1 *makeCombinedRawSpectrum(const TH1 &mb, const TH1 &emc7, double emc7swap, const TH1 &eje, double ejeswap){
+    auto combined = histcopy(&mb);
+    combined->SetDirectory(nullptr);
+    for(auto b : ROOT::TSeqI(combined->GetXaxis()->FindBin(emc7swap), combined->GetXaxis()->FindBin(ejeswap))) {
+        std::cout << "[" << combined->GetXaxis()->GetBinLowEdge(b) << " - " << combined->GetXaxis()->GetBinUpEdge(b) << "] Using EMC7" << std::endl;
+        combined->SetBinContent(b, emc7.GetBinContent(b));
+        combined->SetBinError(b, emc7.GetBinError(b));
+    }
+    for(auto b : ROOT::TSeqI(combined->GetXaxis()->FindBin(ejeswap), combined->GetXaxis()->GetNbins()+1)) {
+        std::cout << "[" << combined->GetXaxis()->GetBinLowEdge(b) << " - " << combined->GetXaxis()->GetBinUpEdge(b) << "] Using EJE" << std::endl;
+        combined->SetBinContent(b, eje.GetBinContent(b));
+        combined->SetBinError(b, eje.GetBinError(b));
+    }
+    return combined;
+}
+
+void runCorrectionChain1DBayes_SpectrumTaskSimplePoor_8TeV(const std::string_view mbfile, const std::string_view emc7file, const std::string_view ejefile, const std::string_view mcfile, const std::string_view sysvar = ""){
+    ROOT::EnableThreadSafety();
+    int NTHREAD=2;
+    std::stringstream outputfile;
+    outputfile << "correctedBayes_poor_austin";
+    if(sysvar.length()) {
+        outputfile << "_" << sysvar;
+    }
+    outputfile << ".root";
+    std::unique_ptr<TFile> mbreader(TFile::Open(mbfile.data(), "READ")),
+                           emc7reader(TFile::Open(emc7file.data(), "READ")),
+                           ejereader(TFile::Open(ejefile.data(), "READ")),
+                           mcreader(TFile::Open(mcfile.data(), "READ")),
+                           writer(TFile::Open(outputfile.str().data(), "RECREATE"));
+    auto binningpart = getJetPtBinningNonLinTruePoor(),
+         binningdet  = getJetPtBinningNonLinSmearPoor(),
+         binningrffine   = getJetPtBinningRejectionFactorsFine(),
+         binningrfcourse = getJetPtBinningRejectionFactorsCourse();
+
+    // Not necessary for 8TeV since trigger cluster "ANY" is used for all triggers
+    //auto centnotrdcorrection = getCENTNOTRDCorrection(*datareader, sysvar);
+
+    double crosssection = 55.8;
+    for(double radius = 0.2; radius <= 0.6; radius += 0.1) {
+        std::cout << "Doing jet radius " << radius << std::endl;
+        auto mbspectrum   = getSpectrumAndNorm(*mbreader, radius, "INT7", "ANY", sysvar),
+             emc7spectrum = getSpectrumAndNorm(*emc7reader, radius, "EMC7", "ANY", sysvar),
+             ejespectrum  = getSpectrumAndNorm(*ejereader, radius, "EJE", "ANY", sysvar);
+        auto trgeffemc7   = makeTriggerEfficiency(*mcreader, radius, "EMC7", sysvar),
+             trgeffeje    = makeTriggerEfficiency(*mcreader, radius, "EJE", sysvar);
```

Comment:
In case you use the "rejection" method you must recall that efficiency losses due to trigger bad channels is already absorbed in the rejection factor while it is not absorbed in the trigger efficiency. Still you need to apply the trigger efficiency, but you need to normalise in the way that it is 1 in the fully-efficient pt-range (high-pt).